src/components/ToggleSwitch.js

```diff
@@ -93,73 +63,103 @@ const ToggleSwitch = React.createClass({
   _renderRightLabel (styles) {
     if (this.props.showLabels) {
       return (
-        <span className='right-label' onClick={this._handleRightLabelClick} style={[styles.label, this.props.labelStyle, this.state.activePosition === 'right' && styles.activeLabel || styles.inactiveLabel]}>{this.props.rightLabel}</span>
+        <span className='right-label' onClick={this._handleToggle} style={[styles.label, this.state.checked && styles.activeLabel || styles.inactiveLabel]}>{this.props.rightLabel}</span>
       );
     } else {
       return null;
     }
   },
 
-  _getChildren () {
-    // Let all the children of the toggle switch be aware of the activePosition
-    return React.Children.map(this.props.children, child => {
-      return React.cloneElement(child, { activePosition: this.state.activePosition });
-    });
+  _renderIcons (styles) {
+    if (!this.props.showIcons) {
+      return null;
+    }
+    let trueIcon = this.props.trueIcon ? this.props.trueIcon : <Icon className='true-icon' style={Object.assign({}, styles.icon, styles.trueIcon)} type='check-skinny' />;
+    let falseIcon = this.props.falseIcon ? this.props.falseIcon : <Icon className='false-icon' style={Object.assign({}, styles.icon, styles.falseIcon)} type='close-skinny' />;
+
+    return (
+      <span>
+        {trueIcon} {falseIcon}
+      </span>
+    );
   },
 
   render () {
-    const styles = {
-      activeLabel: {
-        color: this.props.activeColor
-      },
+    const styles = Object.assign({}, {
       component: {
         display: 'inline-block',
         fontFamily: StyleConstants.FontFamily,
         fontSize: '12px',
         position: 'relative'
       },
-      inactiveLabel: {
-        color: this.props.inactiveColor
+      icon: {
+        fill: StyleConstants.Colors.WHITE,
+        position: 'absolute',
+        top: '0px',
```

Comment:
Don't forget to remove the px on single values like this so that it become `top: 0` instead.